packages/react/src/hooks/useConversationWith.ts

```diff
@@ -0,0 +1,394 @@
+import type TypedEventEmitter from 'typed-emitter';
+import { Room, RoomEvent, ConnectionState, TrackPublishOptions, Track } from 'livekit-client';
+import { EventEmitter } from 'events';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { TokenSource } from '../TokenSource';
+import { useMaybeRoomContext } from '../context';
+import { RoomAgentDispatch, RoomConfiguration } from '@livekit/protocol';
+import { useAgent } from './useAgent';
+import { TrackReference } from '@livekit/components-core';
+import { useLocalParticipant } from './useLocalParticipant';
+
+export enum ConversationEvent {
+  ConnectionStateChanged = 'connectionStateChanged',
+  MediaDevicesError = 'MediaDevicesError',
+}
+
+export type ConversationCallbacks = {
+  [ConversationEvent.ConnectionStateChanged]: (newAgentConnectionState: ConnectionState) => void;
+  [ConversationEvent.MediaDevicesError]: (error: Error) => void;
+};
+
+export type ConversationOptions = {
+  tokenSource: TokenSource;
+  room?: Room;
+
+  dispatch?: {
+    /**
+      * Amount of time in milliseonds the system will wait for an agent to join the room, before
+      * transitioning to the "failure" state.
+      */
+    agentConnectTimeoutMilliseconds?: number;
+  },
+};
+
+export type AgentSessionConnectOptions = {
+  /** Optional abort signal which if triggered will terminate connecting even if it isn't complete */
+  signal?: AbortSignal;
+
+  // FIXME: not sure about this pattern, background thinking is that it would be good to be able to
+  // abstract away enabling relevant media tracks to the caller so they don't have to interface with
+  // the room.
+  tracks?: {
+    microphone?: {
+      enabled?: boolean;
+      publishOptions?: TrackPublishOptions;
+    };
+  };
+};
+
+
+export type SwitchActiveDeviceOptions = {
+  /**
+   *  If true, adds an `exact` constraint to the getUserMedia request.
+   *  The request will fail if this option is true and the device specified is not actually available
+   */
+  exact?: boolean;
+};
+
+type ConversationStateCommon = {
+  subtle: {
```

Comment:
not sure about this naming. what is the intent for exposing it this way?

Reply:
This was largely my ask, Ryan also felt it was a bit awkward.
 
The idea behind it: 
- expose high level APIs directly on the return type, but hide "advanced" use cases behind a common name
- while I agree it doesn't feel super obvious, `subtle` as a name stems from the prior use of it in that context e.g. 
  - https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto
  - https://github.com/tc39/proposal-signals
  - https://pkg.go.dev/crypto/subtle

Reply:
for some of them (e.g. the emitter) I think it would be good to be hidden under a name after discussing this more with @1egoman . 

some alternative suggestions: 
- `advanced`
- `raw`
- `internal`